app/serializers/serialize_exercise_assignment.rb

```diff
@@ -87,9 +88,17 @@ def hints_doc
     Markdown::Render.(solution.git_exercise.hints, :doc)
   end
 
-  def parse_title(header)
-    # Extract everything after the task number
+  def parse_task_title(header)
+    # Parse the task's title, which is everything after the task number
     # "## 2. Do stuff" becomes "Do stuff"
+    # "## General" becomes "General"
     header.to_plaintext.strip.gsub(/^(^\d+)\.\s*(.*)/, '\2')
   end
+
+  def parse_task_id(header)
+    # Parse the task's ID, which is the starting number.
+    # "## 2. Do stuff" becomes 2
+    # "## General" becomes 0
+    header.to_plaintext.strip.gsub(/^(^\d+)\.\s*(.*)/, '\1').to_i
+  end
```

Comment:
I've opted for some duplication, but I could also be converting these into a single method returning a hash 🤷

Reply:
I'm fine with either, although you **could** use the funky Ruby 3 hash-key-pattern matching stuff here probably.